src/church/lists.rs

```diff
@@ -609,6 +609,94 @@ pub fn zip_with() -> Term {
     )
 }
 
+/// Applied to a Church-encoded number `n` and a Church-encoded list it returns the first `n`
+/// elements of the list.
```

Comment:
It's pretty clear that this returns a list, but since I'm already at the review, I'd prefer to be more explicit: "Applied to a Church-encoded number `n` and a Church-encoded list it returns a new list with the first `n` elements of the supplied list."